pkg/skaffold/initializer/init.go

```diff
@@ -538,40 +542,80 @@ func printAnalyzeJSON(out io.Writer, skipBuild bool, pairs []builderImagePair, u
 	return err
 }
 
-func walk(dir string, force, enableJibInit bool, validateBuildFile func(bool, string) ([]InitBuilder, error)) ([]string, []InitBuilder, error) {
+func walk(dir string, force, enableJibInit bool) ([]string, []InitBuilder, error) {
 	var potentialConfigs []string
 	var foundBuilders []InitBuilder
-	err := filepath.Walk(dir, func(path string, f os.FileInfo, e error) error {
-		if f.IsDir() && util.IsHiddenDir(f.Name()) {
-			logrus.Debugf("skip walking hidden dir %s", f.Name())
-			return filepath.SkipDir
-		}
-		if f.IsDir() || util.IsHiddenFile(f.Name()) {
-			return nil
+
+	var enterDirectory func(path string, findBuilders bool) error
+	enterDirectory = func(path string, findBuilders bool) error {
+		dirents, err := godirwalk.ReadDirents(path, nil)
+		if err != nil {
+			return err
 		}
-		if IsSkaffoldConfig(path) {
-			if !force {
-				return fmt.Errorf("pre-existing %s found", path)
+
+		var directories []*godirwalk.Dirent
+		continueSearching := findBuilders
```

Comment:
I like the name `searching`. How about the other way around? Like

`searchBuildersInSubdirectories := findBuilders`

here and

`builderConfigs, continueSearchingBuilders := detectBuilders(...)`
`searchBuildersInSubdirectories = searchBuildersInSubdirectories && continueSearchingBuilders`

below? Because `continueSearching` as of now could give the impression that it is for not breaking out some searching inside the loop. And since we have `foundBuilders`, I find `search` gives more distinguishing feeling for booleans vs builders found. That said, we have `foundBuilders` and `foundConfig`, but the latter is a boolean, so I find it unfortunately too. But not sure what to do about it.

And for `var directories []*godirwalk.Dirent`, how about `subdirectories`?

And for `entryDirectory`, I think something like `searchConfigsAndBuilders` as a recursive search function.

But these are small naming suggestions that I don't feel strongly about, so feel free to go with what you feel comfortable.